docs/rfcs/2022-12-09-ephemeral-conversations.md

```diff
@@ -0,0 +1,208 @@
+
+data ECStateTag
+  = ECSTInvitationSent
+  | ECSTInvitationReceived
+  | ECSTAcptSent
+  | ECSTAcptReceived
+  | ECSTNegotiated
+
+ecStateTag :: ECState -> ECStateTag
+```
+
+\***
+
+Protocol messages:
+
+- `XECInv C.PublicKeyX25519` - invite to ephemeral conversation, other properties except key? ECInvitation type to contain properties?
+
+  - on send: add to Controller's `currentECs` in state `ECInvitationSent` and create `CIECInvitation` chat item.
+
+  - on receive: `processXECInv` - add to Controller's `currentECs` in state `ECInvitationReceived` and create `CIECInvitation` chat item.
+
+- `XECAcpt C.PublicKeyX25519 ConnReqInvitation` - accept ephemeral conversation, send link to join.
+
+  - on send: update Controller's `currentECs` record to state `ECAcptSent`, update chat item.
+
+  - on receive: `processXECAcpt` - update Controller's `currentECs` record to state `ECAcptReceived`, update chat item.
+
+- `XECEnd` - message to end ephemeral conversation. Send in main connection or new one? Main may be better as it may signal cancel as well if ephemeral conversation wasn't yet accepted/negotiated.
+
+Race condition if both parties send `XECInv` simultaneously - if `XECInv` is received when there is ephemeral conversation in `currentECs` in state `ECInvitationSent`, just remove it and signal error `CEECNegotiationError`.
+
+\***
+
+APIs:
+
+- `APIStartEC ContactId` - sends `XECInv`, in UI ephemeral chat view is opened, disabled/progress indicator until ephemeral conversation is negotiated.
+- `APIJoinEC ContactId` - sends `XECAcpt`, in UI ephemeral chat view is opened, disabled/progress indicator until ephemeral conversation is negotiated.
+- `APIOpenEC ContactId` - loads chat items (?) for current ephemeral conversation, opens ephemeral chat view.
+- api to reject? or just allow to delete chat item?
+- `APIEndEC ContactId` - sends `XECEnd`, deletes connection, ephemeral conversation entity and chat items, removes from `currentECs` state, deletes `CIECInvitation` chat item, in UI chat is closed.
+- terminal counterparts
+
+ChatResponses (mirroring chat item updates for terminal):
+
+- `CRECInvitationSent {contact :: Contact}`
+- `CRECInvitationReceived {contact :: Contact}`
+- `CRECAccepted {contact :: Contact}`
+- `CRECAcceptReceived {contact :: Contact}`
+- `CRECEnded {contact :: Contact}`
+
+ChatErrors:
+
+- `CEECNegotiationError {contactId :: ContactId}` - both sent `XECInv`, failed to establish connection, etc.
+- `CENoCurrentEC` - on trying to open, accept, end.
+- `CEECState {currentECState :: ECStateTag}` - on state errors
+
+\***
+
+Chat item content:
+
+``` haskell
+data CIContent (d :: MsgDirection) where
+  ...
+  CIRcvECInvitation ECStateTag -> CIContent 'MDRcv
+  CISndECInvitation ECStateTag -> CIContent 'MDSnd
+  ...
+```
+
+is there a need for more detailed `CIECStatus` or state tag will suffice? (see CICallStatus)
+
+\***
+
+New chat type, direction, chat info:
+
+``` haskell
+data ChatType = ... | CTEphemeral ...
```

Comment:
doesn't seem correct, I think it should be a flag in the contact, not a new chat type.